lib/smb2/file.rb

```diff
@@ -84,34 +105,59 @@ def read(offset: 0, length: nil)
   # @param length [Fixnum] number of bytes to read, starting from `offset`. If
   #   this is greater than the server's maximum read size, the server will
   #   respond with STATUS_INVALID_PARAMETER
-  # @return [String]
-  def read_chunk(offset: 0, length: self.tree.client.max_read_size)
-    packet = Smb2::Packet::ReadRequest.new do |request|
-      request.read_offset = offset
-      request.read_length = length
-      request.file_id = self.create_response.file_id
-      request.minimum_count = 0
-    end
+  # @return [Smb2::Packet::ReadResponse]
+  def read_chunk(offset: self.pos, length: self.tree.client.max_read_size)
+    packet = Smb2::Packet::ReadRequest.new(
+      read_offset: offset,
+      read_length: length,
+      file_id: self.create_response.file_id,
+      minimum_count: 0
+    )
 
     response = tree.send_recv(packet)
 
     response_packet = Smb2::Packet::ReadResponse.new(response)
     @last_read_response = response_packet
 
-    response_packet.data
+    seek(response_packet.data_length, IO::SEEK_CUR)
+
+    response_packet
+  end
+
+  # Seeks to a given `offset` in the stream according to the value of
+  # `whence`:
+  #
+  # |`whence`||
+  # |----------------------|--------------------------------------------------
+  # | :CUR or IO::SEEK_CUR | Seeks to `amount` plus current position
```

Comment:
I'm a bit disinclined to do so because the lines in this table are already quite long. Since it's a table, I can't wrap the lines without breaking it and making them longer will sacrifice readability in the code.